gammapy/spectrum/flux_point.py

```diff
@@ -952,41 +964,41 @@ def estimate_flux_point(self, e_group, steps="all"):
 
         self._set_scale_model()
 
-        if not self.reoptimize:
-            self._freeze_parameters()
+        with self.datasets.parameters.restore_values:
 
-        result.update(self.estimate_norm())
+            self._freeze_empty_background()
 
-        if not result.pop("success"):
-            log.warning(
-                "Fit failed for flux point between {e_min:.3f} and {e_max:.3f},"
-                " setting NaN.".format(e_min=e_min, e_max=e_max)
-            )
+            if not self.reoptimize:
+                self._freeze_parameters()
 
-        if steps == "all":
-            steps = ["err", "errp-errn", "ul", "ts", "norm-scan"]
+            result.update(self.estimate_norm())
 
-        if "err" in steps:
-            result.update(self.estimate_norm_err())
+            if not result.pop("success"):
+                log.warning(
+                    "Fit failed for flux point between {e_min:.3f} and {e_max:.3f},"
+                    " setting NaN.".format(e_min=e_min, e_max=e_max)
+                )
 
-        if "errp-errn" in steps:
-            result.update(self.estimate_norm_errn_errp())
+            if steps == "all":
+                steps = ["err", "errp-errn", "ul", "ts", "norm-scan", "counts"]
```

Comment:
Nitpicking: can you order these in the same way they're run through below?

Reply:
Done